### DISCUSSION TITLE
loadingIndicator

### DISCUSSION THREAD

**Discussion by Cleverson** [prontostack/pronto-fuel#5] [General]
Apresentando o erro Cannot read properties of null (reading 'start')

**Comment by MtDalPizzol** [comment]
Isso ocorre ao navegar entre às páginas? Se sim, garanta que esteja utilizando persistent layouts ou remova o `loadingIndicator` de `resources/frontend/Layouts/Authenticated.vue`. Isso ocorre pq se não usar layouts persistentes, cada página criará uma instância diferente do Layout, destruindo a instância anterior. Então as callbacks responsáveis por mostrar/esconder o indicador de carregamento no Layout terão problemas já que tentarão de ler propriedades de variáveis que já não existem na memória, como nessas linhas:

```js
const loadingIndicator = ref(null)

Inertia.on('start', () => loadingIndicator.value.start())
Inertia.on('finish', () => loadingIndicator.value.stop())
```

**Reply by Cleverson** [reply]
Obrigado pelo feedback, e parabéns pelo conteúdo no youtube e obrigado por disponibilizar uma base para nós
